src/Tasks/Copy.cs

```diff
@@ -290,7 +291,36 @@ private void LogAlwaysRetryDiagnosticFromResources(string messageResourceName, p
                 destinationFileState.FileExists &&
                 !destinationFileState.IsReadOnly)
             {
-                FileUtilities.DeleteNoThrow(destinationFileState.Name);
+                try
+                {
+                    if (NativeMethodsShared.IsLinux)
+                    {
+                        Log.LogMessage($"Run lsof before DeleteNoThrow: {destinationFileState.Name}");
+                        RunLsof();
+                    }
+
+                    Log.LogMessage($"Try to delete with no throw: {destinationFileState.Name}");
+                    FileUtilities.DeleteNoThrow(destinationFileState.Name);
+                }
+                catch (Exception ex) when (ExceptionHandling.IsIoRelatedException(ex))
+                {
+                    Log.LogErrorFromException(ex, showStackTrace: true, showDetail: true, destinationFileState.Name);
+                    if (NativeMethodsShared.IsLinux)
+                    {
+                        Log.LogMessage($"Run lsof before DeleteNoThrow with IsIoRelatedException condition: {destinationFileState.Name}");
```

Comment:
I'm not clear on the benefit of including this in the catches. Either it successfully ran in the try part, in which case this is just duplicate information, or it failed in the try, and there's no reason to suspect it will succeed here?